mlflow/openai/utils.py

```diff
@@ -99,3 +101,56 @@ def request(*args, **kwargs):
             return original(*args, **kwargs)
 
     return _mock_request(new=request)
+
+
+class _OAITokenHolder:
+    def __init__(self, api_type):
+        import openai
+
+        self._api_token = None
+        self._credential = None
+        self._is_azure_ad = api_type in ("azure_ad", "azuread")
+        self._key_configured = bool(openai.api_key) or "OPENAI_API_KEY" in os.environ
+
+        if self._is_azure_ad and not self._key_configured:
+            try:
+                from azure.identity import DefaultAzureCredential
+            except ImportError:
+                raise mlflow.MlflowException(
+                    "Using API type `azure_ad` or `azuread` requires the package"
+                    " `azure-identity` to be installed."
+                )
+            self._credential = DefaultAzureCredential()
+
+    def validate(self, logger=None):
+        """
+        Validates the token or API key configured for accessing the OpenAI resource.
+        """
+        import openai
+
+        if not self._key_configured:
+            if self._is_azure_ad:
+                if not self._api_token or self._api_token.expires_on < time.time() + 60:
+                    from azure.core.exceptions import ClientAuthenticationError
+
+                    if logger:
+                        logger.debug(
+                            "Token for Azure AD is either expired or invalid. Attempting to "
```

Comment:
```suggestion
                            "Token for Azure AD is either expired or unset. Attempting to "
```

`unset` sounds more accurate.